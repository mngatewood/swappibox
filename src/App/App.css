@font-face {
  font-family: "Distant Galaxy";
  src: url("../assets/fonts/SFDistantGalaxy.woff") format("woff");
  font-weight:400;
}

@font-face {
  font-family: "Anakin Mono";
  src: url("../assets/fonts/anakinmono.woff") format("woff");
  font-weight:400;
}

.App {
  text-align: center;
}

header {
  display: flex;
  position: fixed;
  width: 100vw;
  height: 102px;
  top: 0;
  align-content: center;
  background-color: rgba(45, 45, 45, 1);
}

.title-container {
  flex: 1 1 50%;
  padding: 15px 0;
}

.title-container h2 {
  margin: 22px 15px;
  text-align: center;
  font-family: "Distant Galaxy";
  font-size: 42px;
  color: rgba(255, 255, 255, 1);
}

.title-container a {
  text-decoration: none;
}

nav {
  padding: 0;
  flex: 0 0 460px;
  display: inline-block;
  justify-content: space-around;
}

button {
  flex: 1;
  /* margin-top: -10px; */
  width: 100px;
  height: 100px;
  font-size: auto;
  cursor: pointer;
  background-color: rgba(45, 45, 45, 1);
  border: none;
  border-left: 1px lightgray solid;
}

button img {
  height: 30px;
  margin: 22px;
}

button h4 {
  font-size: 18px;
  color: rgba(255, 255, 255, 1);
  margin: -10px 0 15px 0;
}

.count-favorites {
  position: relative;
  top: -10px;
  font-size: 36px;
  width: 50px;
  color: rgba(255, 255, 255, 1);
  cursor: default;
  border: none;
}  

main {
  display: flex;
  /* position: fixed; */
  height: 100vh;
  margin-top: 100px;
  overflow-y: auto;
  /* background-image: url("../assets/images/stars-background.jpg"); */
  background-color: rgba(0, 0, 0, 1);
  color: rgba(255, 255, 255, 1);
}

main h2 {
  margin-left: 300px;
}

aside {
  flex: 0 0 270px;
  position: fixed;
  top: 100px;
  width: 270px;
  height: 100vh;
  padding: 15px;
  /* background-color: rgba(0, 0, 0, 1); */
  overflow: hidden;
}    

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  overflow-y: auto;
  top: 100px;
  margin-left: 300px;
  padding: 15px 0;
  height: auto;
  width: auto;
}
