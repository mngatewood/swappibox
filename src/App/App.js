import React, { Component } from 'react';
import { Route, Switch, Link, NavLink } from 'react-router-dom';
import './App.css';
import { ScrollingText } from '../ScrollingText/ScrollingText'
import Home from '../card-containers/Home'
import People from '../card-containers/People'
import Planets from '../card-containers/Planets'
import Vehicles from '../card-containers/Vehicles'

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      categoriesArray: [],
      // utility: new Utility(),
      film: {},
      people: [],
      planets: [],
      vehicles: [],
      favorites: [],
    };
  }

  componentDidMount() {
    this.updateState();
  }

  updateState = () => {
    this.fetchFilm();
    this.fetchPeople();
  }
  
  fetchFilm = () => {
    const filmIndex = Math.floor(Math.random() * (8 - 1)) + 1;
    const fetchURL = "https://swapi.co/api/films/"
    return fetch(`${fetchURL}${filmIndex}/`)
      .then(response => response.json())
      .then(({ title, opening_crawl: text, release_date: date }) => {
        const film = { title, text, date };
        this.setState({ film })
      })
      .catch(error => ({ error }));
  }

  fetchPeople = () => {
    const fetchURL = "https://swapi.co/api/people/"
    return fetch(fetchURL)
    .then(response => response.json())
    .then(people => (
      this.fetchSpecies(people.results)
    ))
    .then(peopleNameAndSpecies => (
      this.fetchHomeWorlds(peopleNameAndSpecies)
    ))
    .then(peopleNameSpeciesAndHomeWorld => {
      this.setState({people: peopleNameSpeciesAndHomeWorld})
    })
    .catch(error => ({ error }));
  }
  
  fetchSpecies = (people) => {
    const promises = people.map(person => {
      return fetch(person.species)
      .then(response => response.json())
      .then(speciesDetails => ({ 
        name: person.name,
        speciesName: speciesDetails.name, 
        homeWorld: person.homeworld
      }))
      .catch(error => (error));
    });
    return Promise.all(promises);
  };
  
  fetchHomeWorlds = (peopleNameAndSpecies) => {
    const promises = peopleNameAndSpecies.map(person => {
      return fetch(person.homeWorld)
      .then(response => response.json())
      .then(homeWorldDetails => ({
        name: person.name,
        species: person.speciesName,
        homeWorld: homeWorldDetails.name,
        population: homeWorldDetails.population
      }))
      .catch(error => this.setState(error));
    });
    return Promise.all(promises);
  };

  render() {
    return (
      <div className="App">
        <aside>
          <ScrollingText id="scrollMe" film={this.state.film} />
        </aside>
        <main>
            <header className="title-container">
              <Link to='/'><h2>SWAPIbox</h2></Link>
              <button className="view-favorites">View Favorites
                <span className="count-favorites">{this.state.favorites.length}</span>
              </button>       
            </header>
            <section className="button-container">
              <NavLink to='/people'><button className='nav'>People</button></NavLink>
              <NavLink to='/planets'><button className='nav'>Planets</button></NavLink>
              <NavLink to='/vehicles'><button className='nav'>Vehicles</button></NavLink>
            </section>
            <section>
              <Route exact path='/' component={Home} />
              <Route exact path='/people' render={ () => <People 
                people={this.state.people} 
                updateState={this.updateState}
                /> } />
              <Route exact path='/planets' component={Planets} />
              <Route exact path='/vehicles' component={Vehicles} />
            </section>
        </main>
      </div>
    );
  }
}
