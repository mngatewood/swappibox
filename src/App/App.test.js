import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

describe('App', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('should match the snapshot', () => {

  })

  it('should call fetchFilm with no parameters on mount', () => {
    //componentDidMount
    //expect fetchFilm toHaveBeenCalled
  })

  it('should call fetch (film) with correct parameters', () => {
    //instance fetchFilm
    //expect fetch toHaveBeenCalledWith
  })

  it('should display a random opening text once fetched', () => {
    //snapshot after scrolling text is fetched
  })

  it('should call fetchPeople with no parameters on mount', () => {
    //componentDidMount
    //expect fetchPeople toHaveBeenCalledWith
  })

  it('should call fetch (people) with no parameters', () => {
    //instance fetchPeople
    //expect mock fetch toHaveBeenCalledWith
  })

  it('should call fetchSpecies with correct parameters', () => {
    //instance fetchPeople
    //expect fetchSpecies toHaveBeenCalledWith
  })

  it('should call fetch (people.species) with correct parameters', () => {
    //instance fetchSpecies
    //expect mock fetch toHaveBeenCalledWith
  })

  it('should call fetchHomeWorlds with correct parameters', () => {
    //instance fetchSpecies
    //expect fetchHomeWorlds toHaveBeenCalledWith
  })

  it('should call fetch (people.homeworlds) with correct parameters', () => {
    //instance fetchHomeWorlds
    //expect fetch toHaveBeenCalledWith
  })

  it('should update state when fetch is complete', () => {
    //instance fetchHomeWorlds

  })

  it('should display three category buttons', () => {
    //snapshot
  })

  it('should display a view favorites button', () => {
    //snapshot
  })

  it('should default to home page', () => {
    //snapshot
  })

  it('should populate people when button is clicked', () => {
    //
  })

});
