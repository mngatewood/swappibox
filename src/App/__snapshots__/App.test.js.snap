// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <aside>
          <Unknown
                    film={Object {}}
                    id="scrollMe"
          />
</aside>,
        <main>
          <header
                    className="title-container"
          >
                    <Link
                              replace={false}
                              to="/"
                    >
                              <h2>
                                        SWAPIbox
                              </h2>
                    </Link>
                    <button
                              className="view-favorites"
                    >
                              View Favorites
                              <span
                                        className="count-favorites"
                              />
                    </button>
          </header>
          <section
                    className="button-container"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/people"
                    >
                              <button
                                        className="nav"
                              >
                                        People
                              </button>
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/planets"
                    >
                              <button
                                        className="nav"
                              >
                                        Planets
                              </button>
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/vehicles"
                    >
                              <button
                                        className="nav"
                              >
                                        Vehicles
                              </button>
                    </NavLink>
          </section>
          <section>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              exact={true}
                              path="/people"
                              render={[Function]}
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/planets"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/vehicles"
                    />
          </section>
</main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            film={Object {}}
            id="scrollMe"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "film": Object {},
            "id": "scrollMe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="title-container"
>
              <Link
                            replace={false}
                            to="/"
              >
                            <h2>
                                          SWAPIbox
                            </h2>
              </Link>
              <button
                            className="view-favorites"
              >
                            View Favorites
                            <span
                                          className="count-favorites"
                            />
              </button>
</header>,
            <section
              className="button-container"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/people"
              >
                            <button
                                          className="nav"
                            >
                                          People
                            </button>
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/planets"
              >
                            <button
                                          className="nav"
                            >
                                          Planets
                            </button>
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/vehicles"
              >
                            <button
                                          className="nav"
                            >
                                          Vehicles
                            </button>
              </NavLink>
</section>,
            <section>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <Route
                            exact={true}
                            path="/people"
                            render={[Function]}
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/planets"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/vehicles"
              />
</section>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/"
>
                  <h2>
                                    SWAPIbox
                  </h2>
</Link>,
                <button
                  className="view-favorites"
>
                  View Favorites
                  <span
                                    className="count-favorites"
                  />
</button>,
              ],
              "className": "title-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h2>
                    SWAPIbox
</h2>,
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SWAPIbox",
                  },
                  "ref": null,
                  "rendered": "SWAPIbox",
                  "type": "h2",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "View Favorites",
                    <span
                      className="count-favorites"
/>,
                  ],
                  "className": "view-favorites",
                },
                "ref": null,
                "rendered": Array [
                  "View Favorites",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 0,
                      "className": "count-favorites",
                    },
                    "ref": null,
                    "rendered": 0,
                    "type": "span",
                  },
                ],
                "type": "button",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/people"
>
                  <button
                                    className="nav"
                  >
                                    People
                  </button>
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/planets"
>
                  <button
                                    className="nav"
                  >
                                    Planets
                  </button>
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/vehicles"
>
                  <button
                                    className="nav"
                  >
                                    Vehicles
                  </button>
</NavLink>,
              ],
              "className": "button-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": <button
                    className="nav"
>
                    People
</button>,
                  "to": "/people",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "People",
                    "className": "nav",
                  },
                  "ref": null,
                  "rendered": "People",
                  "type": "button",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": <button
                    className="nav"
>
                    Planets
</button>,
                  "to": "/planets",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Planets",
                    "className": "nav",
                  },
                  "ref": null,
                  "rendered": "Planets",
                  "type": "button",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": <button
                    className="nav"
>
                    Vehicles
</button>,
                  "to": "/vehicles",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vehicles",
                    "className": "nav",
                  },
                  "ref": null,
                  "rendered": "Vehicles",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <Route
                  exact={true}
                  path="/people"
                  render={[Function]}
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/planets"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/vehicles"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/people",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/planets",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/vehicles",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "section",
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <aside>
            <Unknown
                        film={Object {}}
                        id="scrollMe"
            />
</aside>,
          <main>
            <header
                        className="title-container"
            >
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    <h2>
                                                SWAPIbox
                                    </h2>
                        </Link>
                        <button
                                    className="view-favorites"
                        >
                                    View Favorites
                                    <span
                                                className="count-favorites"
                                    />
                        </button>
            </header>
            <section
                        className="button-container"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/people"
                        >
                                    <button
                                                className="nav"
                                    >
                                                People
                                    </button>
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/planets"
                        >
                                    <button
                                                className="nav"
                                    >
                                                Planets
                                    </button>
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/vehicles"
                        >
                                    <button
                                                className="nav"
                                    >
                                                Vehicles
                                    </button>
                        </NavLink>
            </section>
            <section>
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                        />
                        <Route
                                    exact={true}
                                    path="/people"
                                    render={[Function]}
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/planets"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/vehicles"
                        />
            </section>
</main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              film={Object {}}
              id="scrollMe"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "film": Object {},
              "id": "scrollMe",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header
                className="title-container"
>
                <Link
                                replace={false}
                                to="/"
                >
                                <h2>
                                                SWAPIbox
                                </h2>
                </Link>
                <button
                                className="view-favorites"
                >
                                View Favorites
                                <span
                                                className="count-favorites"
                                />
                </button>
</header>,
              <section
                className="button-container"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/people"
                >
                                <button
                                                className="nav"
                                >
                                                People
                                </button>
                </NavLink>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/planets"
                >
                                <button
                                                className="nav"
                                >
                                                Planets
                                </button>
                </NavLink>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/vehicles"
                >
                                <button
                                                className="nav"
                                >
                                                Vehicles
                                </button>
                </NavLink>
</section>,
              <section>
                <Route
                                component={[Function]}
                                exact={true}
                                path="/"
                />
                <Route
                                exact={true}
                                path="/people"
                                render={[Function]}
                />
                <Route
                                component={[Function]}
                                exact={true}
                                path="/planets"
                />
                <Route
                                component={[Function]}
                                exact={true}
                                path="/vehicles"
                />
</section>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/"
>
                    <h2>
                                        SWAPIbox
                    </h2>
</Link>,
                  <button
                    className="view-favorites"
>
                    View Favorites
                    <span
                                        className="count-favorites"
                    />
</button>,
                ],
                "className": "title-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h2>
                      SWAPIbox
</h2>,
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "SWAPIbox",
                    },
                    "ref": null,
                    "rendered": "SWAPIbox",
                    "type": "h2",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "View Favorites",
                      <span
                        className="count-favorites"
/>,
                    ],
                    "className": "view-favorites",
                  },
                  "ref": null,
                  "rendered": Array [
                    "View Favorites",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": 0,
                        "className": "count-favorites",
                      },
                      "ref": null,
                      "rendered": 0,
                      "type": "span",
                    },
                  ],
                  "type": "button",
                },
              ],
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/people"
>
                    <button
                                        className="nav"
                    >
                                        People
                    </button>
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/planets"
>
                    <button
                                        className="nav"
                    >
                                        Planets
                    </button>
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/vehicles"
>
                    <button
                                        className="nav"
                    >
                                        Vehicles
                    </button>
</NavLink>,
                ],
                "className": "button-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": <button
                      className="nav"
>
                      People
</button>,
                    "to": "/people",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "People",
                      "className": "nav",
                    },
                    "ref": null,
                    "rendered": "People",
                    "type": "button",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": <button
                      className="nav"
>
                      Planets
</button>,
                    "to": "/planets",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Planets",
                      "className": "nav",
                    },
                    "ref": null,
                    "rendered": "Planets",
                    "type": "button",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": <button
                      className="nav"
>
                      Vehicles
</button>,
                    "to": "/vehicles",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Vehicles",
                      "className": "nav",
                    },
                    "ref": null,
                    "rendered": "Vehicles",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
/>,
                  <Route
                    exact={true}
                    path="/people"
                    render={[Function]}
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/planets"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/vehicles"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "exact": true,
                    "path": "/people",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/planets",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/vehicles",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "section",
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
